R/add_from_raster.R

```diff
@@ -9,29 +9,50 @@
 #' @return A \code{SpatialPointsDataFrame} object.
 #'
 #' @importFrom magrittr %<>%
-#' @importFrom sp spTransform proj4string
+#' @importFrom sp spTransform proj4string SpatialPointsDataFrame
 #' @importFrom raster extract
+#' @importFrom dplyr data_frame
 #'
 #' @export
 #'
 #' @examples
 #' library(sp)
 #' library(magrittr) # for " %>% "
 #'
-#' value <- sptools::gadm("vietnam", "sp", 0) %>%
+#' vn <- sptools::gadm("vietnam", "sp", 0)
+#'
+#' # With a SpatialPointsDataFrame
+#' value <- vn %>%
 #'   spsample(100, "random") %>%
 #'   SpatialPointsDataFrame(data.frame(variable = 1:100)) %>%
 #'   add_from_raster(srtmVN::getsrtm(), "elevation")
 #'
 #' value %>%
 #'   slot("data") %>%
 #'   head()
+#'
+#' # With a SpatialPoints
+#' value <- vn %>%
+#'   spsample(100, "random") %>%
+#'   SpatialPoints(proj4string = vn@proj4string, bbox = vn@bbox) %>%
+#'   add_from_raster(srtmVN::getsrtm(), "elevation")
+#'
+#' value %>%
+#'   slot("data") %>%
+#'   head()
 add_from_raster <- function(sptsdf, rstr, varname = "new_data") {
 # Note that we chose to project the SpatialPointsDataFrame instead of the
 # RasterLayer because it is much quicker this way.
   proj0 <- proj4string(sptsdf)
   sptsdf %<>% spTransform(proj4string(rstr))
-  sptsdf[[varname]] <- raster::extract(rstr, sptsdf)
+  if (class(sptsdf) == "SpatialPoints") {
+    sptsdf <- SpatialPointsDataFrame(sptsdf@coords,
+                                     data_frame(!!varname :=
+                                                raster::extract(rstr, sptsdf)),
+                                     proj4string = rstr@crs)
```

Comment:
In terms of reading consistency, wouldn't it make more sense to write `proj4string = proj4string(sptsdf)` instead of `proj4string = rstr@crs`?

Reply:
I agree, I will do it now